subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoTaskExtension.java

```diff
@@ -290,7 +294,7 @@ public String getAsJvmArg() {
         builder.append("-javaagent:");
         builder.append(GFileUtils.relativePath(task.getWorkingDir(), agent.getJar()));
         builder.append('=');
-        argument.append("destfile", getDestinationFile());
+        argument.append("destfile", destinationFile.isPresent() ? getDestinationFile() : null);
```

Comment:
given this pattern appears a bunch of time, it might be good to add something like `Provider.getOrNull()`